test/libdnf/base/test_goal.cpp

```diff
@@ -53,6 +54,36 @@ void BaseGoalTest::test_install() {
     CPPUNIT_ASSERT_EQUAL(0lu, obsoleted_set.size());
 }
 
+void BaseGoalTest::test_install_not_available() {
+    libdnf::Goal goal(base.get());
+    base->get_config().strict().set(libdnf::Option::Priority::RUNTIME, false);
+    base->get_config().best().set(libdnf::Option::Priority::RUNTIME, true);
+    goal.add_rpm_install("not_available");
+    goal.resolve(false);
+    auto install_set = goal.list_rpm_installs();
+    auto reinstall_set = goal.list_rpm_reinstalls();
+    auto upgrade_set = goal.list_rpm_upgrades();
+    auto downgrade_set = goal.list_rpm_downgrades();
+    auto remove_set = goal.list_rpm_removes();
+    auto obsoleted_set = goal.list_rpm_obsoleted();
+    CPPUNIT_ASSERT_EQUAL(0lu, install_set.size());
+    CPPUNIT_ASSERT_EQUAL(0lu, reinstall_set.size());
+    CPPUNIT_ASSERT_EQUAL(0lu, upgrade_set.size());
+    CPPUNIT_ASSERT_EQUAL(0lu, downgrade_set.size());
+    CPPUNIT_ASSERT_EQUAL(0lu, remove_set.size());
+    CPPUNIT_ASSERT_EQUAL(0lu, obsoleted_set.size());
+
+    auto & log = goal.get_resolve_log();
+    CPPUNIT_ASSERT_EQUAL(1lu, log.size());
+    auto & [action, problem, settings, spec, additional_data] = *log.begin();
```

Comment:
Just for the record (it's unrelated to this PR) - calling this `log` confused me.
It uses underlying `add_rpm_goal_report()` method.
If I understand what it should be doing correctly, it should be called `rpm_goal_job_result` or something like that.

Reply:
I know that this is confusing, but I suggest that it will be more clear after implementation of modularity (on API and code behind that). I suggest that modularity will also generate resolve reports, but I cannot tell if it will be compatible with other logs.